### ISSUE TITLE
What is the usage of MS.NETCore.App.Host 3.1 and 6.0 in SDK and can it be removed for source-build?

### ISSUE THREAD

**Issue by crummel** [dotnet/sdk#26482]
In source-build we see prebuilt versions 3.1 and 6.0 of Microsoft.NETCore.App.Host being used.  What bring these in?  Is it running bits targeted for net6.0 in the source-build build?  Should we be retargeting everything to net7?

**Comment by MichaelSimons** [comment]
cc @dsplaisted

**Comment by marcpopMSFT** [comment]
@crummel , where do you see the downlevel app-hosts being used in source build as I don't think they need to be? When building a framework dependent application we create a foo.dll with the content of your application and then an executable with the same name. This executable is [architecture and platform specific](https://www.nuget.org/packages?q=Microsoft.NETCore.App.Host) and so we use existing app hosts (generated by the runtime) to create those executables. 

On windows, we include all architectures with the SDK and VS installs so apps don't have to download those on the fly but if you're targeting linux from a windows machine (for a framework-dependent application) we will download the apphost from your feed. We don't include downlevel app hosts though (we do in VS but those come from the matching runtime, not the SDK).

TLDR: I wouldn't expect downlevel apphosts included in source build.

**Comment by MichaelSimons** [comment]
Taking another at these, I don't see that they are coming from the sdk.  I will investigate further.
